dbclient/HiveClient.py

```diff
@@ -365,14 +373,19 @@ def import_hive_metastore(self, cluster_name=None, metastore_dir='metastore/', v
                 tables = self.listdir(local_db_path)
                 for tbl_name in tables:
                     # build the path for the table where the ddl is stored
-                    print("Importing table {0}.{1}".format(db_name, tbl_name))
-                    local_table_ddl = metastore_local_dir + db_name + '/' + tbl_name
-                    if not self.move_table_view(db_name, tbl_name, local_table_ddl):
-                        # we hit a table ddl here, so we apply the ddl
-                        is_successful = self.apply_table_ddl(local_table_ddl, ec_id, cid, db_path, has_unicode)
-                        print(is_successful)
+                    full_table_name = f"{db_name}.{tbl_name}"
+                    if checkpoint_metastore_set.contains(full_table_name):
+                        print(f"Table {full_table_name} found in checkpoint file, already imported")
                     else:
-                        print(f'Moving view ddl to re-apply later: {db_name}.{tbl_name}')
+                        print(f"Importing table {full_table_name}")
+                        local_table_ddl = metastore_local_dir + db_name + '/' + tbl_name
+                        if not self.move_table_view(db_name, tbl_name, local_table_ddl):
+                            # we hit a table ddl here, so we apply the ddl
+                            is_successful = self.apply_table_ddl(local_table_ddl, ec_id, cid, db_path, has_unicode)
+                            checkpoint_metastore_set.write(full_table_name)
```

Comment:
Should we log the checkpoint when is_successful == false?

Reply:
Not changed?

Reply:
Actually, apply_table_ddl function is just returning the response and was not handling errors, added to logic to checkpoint only on success